src/gsCore/gsDofMapper.cpp

```diff
@@ -282,6 +295,42 @@ void gsDofMapper::setIdentity(index_t nPatches, size_t nDofs)
     m_dofs.resize( m_numFreeDofs, 0);
 }
 
+void gsDofMapper::permuteFreeDofs(const gsVector<index_t>& permutation)
+{
+    GISMO_ASSERT(m_curElimId==0, "finalize() was not called on gsDofMapper");
+    GISMO_ASSERT(m_numFreeDofs == permutation.size(), "permutation size does not match number of free dofs");
+    //GISMO_ASSERT(m_tagged.empty(), "you cannot permute the dofVector twice, combine the permutation");
+    //We could check here, that permutation is indeed a permutation
+
+    //make a copy of the old ordering, easiest way to implement the permutation. Inplace reordering is quite hard.
+    std::vector<index_t> dofs = m_dofs;
+    //first use a tempory coupled vector, other wise is_coupled_index() would not work correctly
+    std::vector<index_t> tagged;
+    tagged.reserve(taggedSize()); //reserve enough memory
```

Comment:
Would be better if tagged where not changed inside this function..
You could do in your codes:

mapper.markCoupledAsTagged();
mapper.permuteFreeDofs(..)

and that is all

Reply:
This is then related to the comment above

Reply:
OK: this made me a little confused.
You could rename tagged --> tagged_permuted for clarity

Reply:
ok